packages/core/admin/admin/src/content-manager/components/DynamicTable/index.js

```diff
@@ -90,40 +97,51 @@ const DynamicTable = ({
     return formattedHeaders;
   }, [runHookWaterfall, displayedHeaders, layout, hasDraftAndPublish, formatMessage]);
 
+  const withMainAction = (canDelete || canPublish) && isBulkable;
+  const rowCount = rows.length + 1;
+  // Add 1 for the visually hidden actions header, and 1 for select all checkbox if the table is bulkable
+  const colCount = tableHeaders.length + 1 + (withMainAction ? 1 : 0);
+
   return (
-    <Table
-      components={{ ConfirmDialogDelete }}
-      contentType={contentTypeName}
-      action={action}
-      isLoading={isLoading}
-      headers={tableHeaders}
-      onConfirmDelete={onConfirmDelete}
-      onOpenDeleteAllModalTrackedEvent="willBulkDeleteEntries"
-      rows={rows}
-      withBulkActions
-      withMainAction={(canDelete || canPublish) && isBulkable}
-      renderBulkActionsBar={({ selectedEntries, clearSelectedEntries }) => (
-        <BulkActionsBar
-          showPublish={canPublish && hasDraftAndPublish}
-          showDelete={canDelete}
-          onConfirmDeleteAll={onConfirmDeleteAll}
-          onConfirmPublishAll={onConfirmPublishAll}
-          onConfirmUnpublishAll={onConfirmUnpublishAll}
-          selectedEntries={selectedEntries}
-          clearSelectedEntries={clearSelectedEntries}
-        />
+    <TableProvider rowCount={rowCount} colCount={colCount}>
+      {isLoading || !rows.length ? (
+        <Table rowCount={rowCount} colCount={colCount} shadow="">
+          <TableEmptyBody
+            colSpan={colCount}
+            action={action}
+            contentType={contentTypeName}
+            isLoading={isLoading}
+          />
+        </Table>
+      ) : (
+        <>
+          <TableActionBar>
+            <BulkActionsBar
+              showPublish={canPublish && hasDraftAndPublish}
+              showDelete={canDelete}
+              onConfirmDeleteAll={onConfirmDeleteAll}
+              onConfirmPublishAll={onConfirmPublishAll}
+              onConfirmUnpublishAll={onConfirmUnpublishAll}
+            />
+          </TableActionBar>
+          <Table rowCount={rowCount} colCount={colCount}>
```

Comment:
This is the Table from the DS. It didn't make sense to me to have the ds Table in the helper plugin since it would basically just be re-exporting the same component and fowarding the props. wdyt?